kong/plugins/zipkin/handler.lua

```diff
@@ -205,6 +213,14 @@ elseif subsystem == "stream" then
 
     request_span:set_tag("lc", "kong")
 
+    local static_tags = conf.static_tags
+    if type(static_tags) == "table" then
+      for i = 1, #static_tags do
```

Comment:
How do you choose between this technique and using `ipairs`?

Reply:
This one I'm not sure but if you do get confused between pairs and ipairs, we use `ipairs` in the hot code paths to increase the probability that it is compiled by the JIT: http://wiki.luajit.org/NYI

Reply:
Performance wise, both alternatives have similar speed (both get JITted) in LuaJIT. I think `ipairs` is more readable, so if you are sure your code is going to get executed exclusively in LuaJIT, you should use ipairs.

However, if there's a chance your code is *not* going to be executed in LuaJIT, the performance of `ipairs` is way worse than the numeric alternative in Vanilla Lua.

Since I maintain several non-LuaJIT-specific open source Lua libraries I *always* use the numeric `for` "on automatic" - I almost never use `ipairs`. I had internalized this so much that I had to think and reevaluate my thought process :)